src/www/app/util.ts

```diff
@@ -14,14 +14,14 @@
 
 import * as errors from '../model/errors';
 
-export function timeoutPromise<T>(promise: Promise<T>, ms: number, name = ''): Promise<T> {
-  let winner: Promise<T>;
+export function timeoutPromise<T>(promise: Promise<T>, ms: number, name = ''): Promise<T | undefined> {
+  let winner: Promise<T | undefined> | undefined = undefined;
   const timeout = new Promise<T>((resolve, reject) => {
     const timeoutId = setTimeout(() => {
       clearTimeout(timeoutId);
-      if (winner) {
+      if (winner === undefined) {
```

Comment:
This reverses the polarity of the check ... but actually the check is totally broken already.  (This branch would never be taken, and now will always be taken.)

Given that (evidently) we are not actually using these log messages, I propose getting rid of them, which makes this function much simpler.

Reply:
Ah, derp, it should be `!== undefined`. Switched. But yeah I was like -- wait is this even needed?

Reply:
Also evidently this function has no test coverage!

Reply:
Test coverage and automation I think is the next big dev project for me anyway.

I think this has the same intent? Sanity check me:

```js
export function promiseTimeout<T>(promise: Promise<T>, timeoutDuration: number, timeoutName: string) {
  return Promise.race([
    promise,
    new Promise((_, reject) =>
      setTimeout(() => reject(new errors.OperationTimedOut(timeoutDuration, timeoutName)), timeoutDuration)
    ),
  ]);
}
```

Reply:
We're only using this in one place, btw

https://github.com/Jigsaw-Code/outline-client/blob/1b1bdcf416f9db09dc655d4d43e03d9e11177b78/src/electron/connectivity.ts#L28-L29

Reply:
I think you still want the `<T>`, but otherwise yes.

Reply:
Alright sure. Pushed.